utils/execTo.js

```diff
@@ -0,0 +1,23 @@
+const {
+  execSync
+} = require(`child_process`);
+
+const execToStr = (cmd) => {
+  var _execSync;
+
+  return String((_execSync = execSync(cmd, {
+    encoding: `utf8`
+  })) !== null && _execSync !== void 0 ? _execSync : ``).trim();
+}
```

Comment:
This is a little unclear to me. I think we're trying to do a little too much in this return statement. Let me see if I can break this apart:

`String(...` We're going to make a string.

`_execSync = execSync(` We're setting the value returned by `child_process.execSync` to a variable inline - we probably don't want to do so.

`execSync(cmd, { encoding: 'utf8' })` We're getting a string that's the result of running this command. I think we probably want the `encoding` on the _String_ and not the _execSync_.

`!== null && _execSync !== void 0` If the string isn't null (I don't think it could be) and the command's output isn't void (not sure how that works).

`? _execSync : ''` set the string to be the results _or an empty string.

`).trim();` Remove empty spaces.

---

If I'm following that correctly, then this might be able to be simplified to the following:

```js
// get the output of the execSync command
const result = execSync(cmd);

// If the result is null, void, or any other falsy value, return an empty string
if (!result) {
  return '';
}

// Otherwise, return the results with the correct encoding
return String(result, { encoding: 'utf8' });
```

Reply:
There's a great talk by Rich Hickey (the creator of Clojure) called "Simple Made Easy". Untangling and simplifying this code made me think of it.

Reply:
OG Typescript: will test 🤠 
definitely going to give that a look